lib/DxilPIXPasses/DxilPixEmitMetadata.cpp

```diff
@@ -0,0 +1,43 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxilPixEmitResources.cpp                                                  //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Emit Dxil resource meta data                                              //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxc/DXIL/DxilFunctionProps.h"
+#include "dxc/DXIL/DxilModule.h"
+#include "dxc/DXIL/DxilOperations.h"
+#include "dxc/DXIL/DxilUtil.h"
+#include "dxc/DxilPIXPasses/DxilPIXPasses.h"
+#include "dxc/DxilPIXPasses/DxilPIXVirtualRegisters.h"
+#include "dxc/HLSL/DxilGenerationPass.h"
+
+#include "llvm/IR/Module.h"
+
+using namespace llvm;
+using namespace hlsl;
+
+class DxilPixEmitMetadata : public ModulePass {
+public:
+  static char ID;
+  explicit DxilPixEmitMetadata() : ModulePass(ID) {}
+  bool runOnModule(Module &M) override {
+    DxilModule &DM = M.GetOrCreateDxilModule();
+    DM.ReEmitDxilResources();
+    return true;
```

Comment:
Is "true" appropriate here, given that only the metadata is changed?

Reply:
I believe that counts!